docroot/modules/custom/mass_content/src/Commands/MassContentCommands.php

```diff
@@ -456,4 +458,82 @@ public function getEntityBundles($type) {
     return $bundles;
   }
 
+  /**
+   * Search and replace media URLs.
+   *
+   * @command mass-content:search-replace-media
+   *
+   * @aliases msrm
+   */
+  public function searchAndReplaceMedia() {
+    $entityTypes = ['node', 'paragraph'];
+    foreach ($entityTypes as $entityType) {
+      // Retrieve the last processed ID from the state or default to 0
+      $lastProcessedId = \Drupal::state()->get("mass_content.last_processed_id.{$entityType}", 0);
+
+      // Determine the ID field name based on the entity type
+      $idFieldName = $entityType === 'node' ? 'nid' : 'id';
+      $storage = $this->entityTypeManager->getStorage($entityType);
+      $ids = $storage->getQuery()
+        ->condition($idFieldName, $lastProcessedId, '>')
+        ->sort($idFieldName)
+        ->accessCheck(FALSE)
+        ->execute();
+
+      $urlReplacementService = \Drupal::service('mass_fields.url_replacement_service');
+      $changedEntities = 0;
+      foreach ($ids as $id) {
+        // After successfully processing the entity, update the last processed ID
+        \Drupal::state()->set("mass_content.last_processed_id.{$entityType}", $id);
+        $entity = $storage->load($id);
+        if ($entity) {
+          $changed = FALSE;
+          foreach ($entity->getFields() as $field) {
+            $fieldType = $field->getFieldDefinition()->getType();
+            if (in_array($fieldType, ['text_long', 'text_with_summary', 'string_long'])) {
+              foreach ($field as $item) {
+                $processed = $urlReplacementService->processText($item->value);
+                if ($processed['changed']) {
+                  $item->value = $processed['text'];
+                  $changed = TRUE;
+                }
+              }
+            }
+          }
+          if ($changed) {
+            if (method_exists($entity, 'setRevisionLogMessage')) {
+              $entity->setNewRevision();
+              $entity->setRevisionLogMessage('Revision created to fix raw media/ or files/ URL in the content.');
+              $entity->setRevisionCreationTime(\Drupal::time()
+                ->getRequestTime());
+            }
+            $entity->save();
```

Comment:
@arthurbaghdas Just to sanity check, is there a chance that authors could receive watch notifications?

Reply:
Forgot to add the snippet. thanks for reminding.